core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BpmBillingProjectLifecycle.scala

```diff
@@ -119,7 +119,10 @@ class BpmBillingProjectLifecycle(
       }
 
       val maybeAttach = if (lzId.isDefined) Map("attach" -> "true") else Map.empty
-      val params = config.azureConfig.get.landingZoneParameters ++ maybeAttach
+      val costSavingParams =
+        if (createProjectRequest.costSavings.getOrElse(false)) config.azureConfig.get.costSavingLandingZoneParameters
```

Comment:
it seems like that if the costSaving bool is included but is set to false, this code will still use the cost savings params that are hardcoded to true, or am I reading this wrong?

Reply:
the full line is 
`val costSavingParams = if (createProjectRequest.costSavings.getOrElse(false)) config.azureConfig.get.costSavingLandingZoneParameters else Map.empty`
so there's an else that sets it to empty like the `maybeAttach` val above it

Reply:
Scala syntax goodness: using `contains` might be easier to read because it mentions the word "true":
```scala
if (createProjectRequest.costSavings.contains(true))
```
no functional difference, and this is just style, so your call to change or not

Reply:
I see, the value get set to false if we cant get it, which means the if will not evaluate to true. I was confusing the getorElse and if/else together. thanks for clearing it up :)